src/lightning/fabric/plugins/io/xla.py

```diff
@@ -61,4 +62,5 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], path: _PATH, storage_optio
             checkpoint = apply_to_collection(checkpoint, (DictConfig, ListConfig), OmegaConf.to_container)
         import torch_xla.core.xla_model as xm
 
-        xm.save({k: v for k, v in checkpoint.items() if k != "callbacks"}, path)
+        cpu_data = xm._maybe_convert_to_cpu(checkpoint, convert=True)
+        torch.save(cpu_data, path)
```

Comment:
I replaced `xm.save` because this logic: https://github.com/pytorch/xla/blob/080a70d2ed16150d6e00c6d7adb079aa6cb58a4f/torch_xla/utils/serialization.py#L71-L72 does not work well when one does `devices=[1]` because then nothing gets saved